javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistRecordDeclarationTest.java

```diff
@@ -322,6 +330,35 @@ void testNonCanonicalConstructor() {
         assertEquals("java.lang.String", stringConstructor.getParam(0).getType().describe());
     }
 
+    @Test
+    @EnabledForJreRange(min = org.junit.jupiter.api.condition.JRE.JAVA_17)
+    void genericConstructorTest() {
```

Comment:
It seems to me that this is not a Javassist test.

Reply:
The reference to the jar is through the typeSolver. box.GenericBox is imported, but this declaration is included in the Box.jar used to create the type solver.


There's a similar test for the reflection model